File: week05/echon.pl
#!/usr/bin/perl -w

if ($#ARGV + 1 != 2) {
   print STDERR "Usage: ./echon.pl <number of lines> <string>\n";
   exit 1;
} elsif($ARGV[0] !~ /^\d+$/) {
   print STDERR "$0: argument 1 must be a non-negative integer\n";
   exit 1;
}


for (1..$ARGV[0]) {
   print "$ARGV[1]\n";
}

File: week06/echon.py
#!/usr/bin/python3

import re
import sys

if len(sys.argv) != 3:
   print("Usage: %s <number of lines> <string>" %(sys.argv[0]), file=sys.stderr)
   exit(1)

if not re.search(r'^\d+$', sys.argv[1]):
   print("%s: argument 1 must be a non-negative integer" %(sys.argv[0]), file=sys.stderr)
   exit(1)

n = int(sys.argv[1])
for i in range(0,n):
   print("%s" %(sys.argv[2]))
